src/renderer/dx/DxRenderer.cpp

```diff
@@ -802,17 +939,37 @@ void DxEngine::SetCallback(std::function<void()> pfn)
     _pfn = pfn;
 }
 
-bool DxEngine::GetRetroTerminalEffects() const noexcept
+void DxEngine::SetWarningCallback(std::function<void(const HRESULT)> pfn)
+{
+    _pfnWarningCallback = pfn;
+}
+
+bool DxEngine::GetRetroTerminalEffect() const noexcept
 {
-    return _retroTerminalEffects;
+    return _retroTerminalEffect;
 }
 
-void DxEngine::SetRetroTerminalEffects(bool enable) noexcept
+void DxEngine::SetRetroTerminalEffect(bool enable) noexcept
 try
 {
-    if (_retroTerminalEffects != enable)
+    if (_retroTerminalEffect != enable)
     {
-        _retroTerminalEffects = enable;
+        // The caller might want to leave the shader
+        _terminalEffectsEnabled = true;
+        _retroTerminalEffect = enable;
+        _recreateDeviceRequested = true;
+        LOG_IF_FAILED(InvalidateAll());
+    }
+}
+CATCH_LOG()
+
+void DxEngine::SetPixelShaderPath(std::wstring_view value) noexcept
+try
+{
+    if (_pixelShaderPath != value)
+    {
+        _terminalEffectsEnabled = true;
+        _pixelShaderPath = { value };

```

Comment:
How do we set effects to false? neither setter with False disables it..

Reply:
.... That's a good point